models/_generate_layers.py

```diff
@@ -496,6 +497,38 @@ def collect_layers():
   return ls
 
 
+def get_super_call_params(layer):
+  """
+  Inspects the given layer, extracts unwanted parameters and then returns a string that is ready to be used in the super
+  call.
+  :param layer: Layer to be inspected
+  :return: string with super call parameters
+  """
+  # get code as string list
+  code = inspect.getsource(layer.__init__).splitlines()
+
+  # get list of lines starting with super and take the first one
+  lines = [line.strip() for line in code if line.strip().startswith("super")]
+  if not lines:
+    return ''
+
+  # reformat the super call to extract what we need
+  super_call = lines[0]
+  call = super_call.split(".")[1]
+  call_pruned = call[9:-1]
```

Comment:
Never use such hardcoded values like `9`. What is 9?

Reply:
Also, I assume that you have certain assumptions what the first 9 chars, and the last 1 char of `call` is, right? If you have such assumptions, then always also `assert` for them.

Reply:
9 is the length of `"__init__(" and 1 is the length of ")", in the next update of commits I will push this will be clarified through comments and also I replaced the two hardcoded values by `len("__init__(")` and `len(")") in the statement, which makes the code more understandable, but a bit more messy I guess. 
Would be good to hear what you think what we should put here.

Reply:
Code must be understandable. This is a hard requirement.
Also, understandable code is it less messy, not more messy.